packages/react-components/src/icons/src/Icon.tsx

```diff
@@ -1,30 +1,43 @@
 import "./Icon.css";
 
 import { Box } from "../../box";
-import { cssModule, mergeProps, normalizeSize, slot, useStyleProps } from "../../shared";
-import { elementType, oneOf, string } from "prop-types";
-import { forwardRef } from "react";
+import { ComponentProps, ElementType, ForwardedRef, ReactElement } from "react";
+import { OrbitComponent, cssModule, forwardRef, mergeProps, normalizeSize, slot, useStyleProps } from "../../shared";
 import { isNil } from "lodash";
 
-const propTypes = {
-    /**
-     * An icon as a React component.
-     */
-    type: elementType.isRequired,
+export interface CreatedIconProps {
     /**
      * An icon can vary in size.
      */
-    size: oneOf(["2xs", "xs", "sm", "md", "lg", "xl", "inherit"]),
+    size?: "2xs" | "xs" | "sm" | "md" | "lg" | "xl" | "inherit";
     /**
      * An icon can change inherit it's parent color.
      */
-    color: oneOf(["inherit"]),
+    color?: "inherit";
     /**
      * Default slot override.
      */
-    slot: string
-};
-export function InnerIcon(props) {
+    slot?: string;
+}
+
+export interface InnerIconProps extends CreatedIconProps {
+    /**
+     * An icon as a React component.
```

Comment:
I would document this as "the concrete icon type" something like this.

```
import { Icon } from "@orbit-ui/react-components";
import { ReactComponent as Azure32 } from "./icon-azure-32.svg";

<Icon type={Azure32} />
```

Reply:
In fact, since we use "as" everywhere, should this be renamed to "as" ?

Reply:
Wouldn't work with MultiVariantIcons